### PR TITLE
Deploy minio operator in integreatly ocp4 workload

### PR DESCRIPTION
**Pull request by dimitraz**
##### SUMMARY
Deploys the minio operator as part of the pre-workload tasks
https://issues.redhat.com/browse/INTLY-7040

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: ansible/roles/ocp4-workload-integreatly/defaults/main.yml [hunk 1]
```diff
@@ -21,7 +21,17 @@ silent: False
 # (if defined).
 #
 # The logic in the workload should only use the dictionary "<role name>".
```
Comment (aidenkeating) [inline ansible/roles/ocp4-workload-integreatly/defaults/main.yml:23]: Based on this comment, should these variables all be under the `ocp4_workload_integreatly_defaults` dict?

Comment (dimitraz) [reply ansible/roles/ocp4-workload-integreatly/defaults/main.yml:23]: I didn't think we'd want those as override params :thinking:

Comment (aidenkeating) [reply ansible/roles/ocp4-workload-integreatly/defaults/main.yml:23]: Yep, ignore

**Comment by dimitraz** [comment]
@wkulhanek @pmccarthy mind taking a look when you get a chance?

File: ansible/roles/ocp4-workload-integreatly/defaults/main.yml [hunk 2]
```diff
@@ -3,6 +3,19 @@ become_override: False
 ocp_username: opentlc-mgr
 silent: False
 
+# variables for the minio operator deployment
+minio_operator_ns: "redhat-rhmi-minio-operator"
```
Comment (wkulhanek) [inline ansible/roles/ocp4-workload-integreatly/defaults/main.yml:7]: If these are part of the Integreatly workload would you mind sticking these variables into the dictionary?
```
ocp4_workload_integreatly_defaults:
  minio_operator_ns
  ...
```

File: ansible/roles/ocp4-workload-integreatly/templates/clusterrole.j2 [hunk 3]
```diff
@@ -0,0 +1,52 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ minio_operator_role }}
```
Comment (wkulhanek) [inline ansible/roles/ocp4-workload-integreatly/templates/clusterrole.j2:4]: And of course here you'd use `{{ ocp4-workload-integreatly.minio_operator_role }}`

File: ansible/roles/ocp4-workload-integreatly/defaults/main.yml (hunk 2 above)
Comment (wkulhanek) [reply ansible/roles/ocp4-workload-integreatly/defaults/main.yml:7]: But really I'd stick that into its own workload. Something like ocp4-workload-integreatly-minio. And then run that before running the integreatly workload.

Comment (dimitraz) [reply ansible/roles/ocp4-workload-integreatly/defaults/main.yml]: Sure, thanks for the feedback. We've updated this now to a separate workload

File: ansible/configs/ocp-workloads/ocp-workload.yml [hunk 4]
```diff
@@ -17,7 +17,17 @@
...
+
+  - name: Run multiple workloads
+    when:
+      - ocp_workloads is defined
+      - ocp_workloads | length > 0
+    loop: "{{ ocp_workloads }}"
+    loop_control:
+      loop_var: _ocp_workload
+    include_role:
+      name: "{{ _ocp_workload }}"
```
Comment (aidenkeating) [inline ansible/configs/ocp-workloads/ocp-workload.yml:33]: @wkulhanek Do you see an issue with having something like this in here? It allows us to test overriding `ocp_workloads` as well as a single `ocp_workload`. Which is useful as for local dev we want to run Minio before the main `integreatly` role

File: ansible/roles/ocp4-workload-integreatly/sample_vars/ocp4_workload_integreatly_vars.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+# override ocp workloads to include prerequisite roles
+ocp_workloads:
+  - ocp4-workload-integreatly-minio
+  - ocp4-workload-integreatly
```
Comment (aidenkeating) [inline ansible/roles/ocp4-workload-integreatly/sample_vars/ocp4_workload_integreatly_vars.yml:5]: Wasn't sure where to put this, but it's useful for local dev and it's the order we'll need the roles run in in the actual workshop on RHPDS.

File: ansible/configs/ocp-workloads/ocp-workload.yml (hunk 4 above)
Comment (wkulhanek) [reply ansible/configs/ocp-workloads/ocp-workload.yml:33]: @aidenkeating That's a good idea. Had written up a whole thing.... until I realized that the variable is different( workloads vs workload) (plural vs singular). I think this is good.

**Pull request merged**